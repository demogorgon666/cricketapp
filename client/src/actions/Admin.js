import {
  GET_PLAYERS,
  CREATE_PLAYER,
  CREATE_TEAM,
  EDIT_PLAYER,
  EDIT_TEAM,
  DELETE_PLAYER,
  DELETE_TEAM,
  GET_PLAYER_SEARCH,
  GET_TEAM_SEARCH,
  GET_ALLTEAMS,
  ERROR_TYPE
} from "./Types";
import axios from "axios";

// export const getPlayers = () => dispatch => {
// 	return axios
// 		.get("http://localhost:5000/api/cricketalpha/allPlayer")
// 		.then(res => {
// 			console.log("From Player Actions", res.data.data);
// 			dispatch({
// 				type: GET_PLAYERS,
// 				payload: res.data.data
// 			});
// 		})
// 		.catch(err => {
// 			console.log(err);
// 		});
// };
//
export const getPlayers = () => dispatch => {
  console.log(localStorage.getItem("token"));
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/allPlayer", {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: GET_PLAYERS,
        payload: res.data.data
      });
      console.log(res.data);
    })
    .catch(err => {
      console.log(err);
    });
};

// export const getTeams = () => dispatch => {
//     return axios
//       .get("http://localhost:5000/api/cricketalpha/allteams")
//       .then(res => {
//         dispatch({
//           type: GET_TEAMS,
//           payload: res.data.data
//         });
//       })
//       .catch(err => {
//         console.log(err);
//       });
//   };

export const getAllTeams = () => dispatch => {
  console.log(localStorage.getItem("token"));
  return axios
    .get("http://localhost:5000/apis/admin/allteam", {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: GET_ALLTEAMS,
        payload: res.data.data
      });
      console.log(res.data);
    })
    .catch(err => {
      console.log(err);
    });
};

//   export const createPlayer = player => dispatch => {
//     return axios
//       .post("http://localhost:5000/api/cricketalpha/player/new", player)
//       .then(res => {
//         dispatch({
//           type: CREATE_PLAYER
//         });
//         console.log("User created successfully");
//       })
//       .catch(err => {
//         console.log(err);
//       });
//   };

//   export const createTeam = team => dispatch => {
//     return axios
//       .post("http://localhost:5000/api/cricketalpha/user/new", team)
//       .then(res => {
//         dispatch({
//           type: CREATE_TEAM
//         });
//         console.log("User created successfully");
//       })
//       .catch(err => {
//         console.log(err);
//       });
//   };
export const createTeam = team => dispatch => {
  return axios
    .post("http://localhost:5000/apis/admin/team/new", team, {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: CREATE_TEAM
      });
      alert("Added one team successfully");
      dispatch(getAllTeams());
    })
    .catch(err => {
      console.log(err);
      dispatch({ type: ERROR_TYPE, payload: err.response.data.message });
      console.log(err.response.data.message);
    });
};

export const createPlayer = player => dispatch => {
  return axios
    .post("http://localhost:5000/apis/admin/player/new", player, {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: CREATE_PLAYER
      });
      alert("Added one player successfully");
      dispatch(getPlayers());
    })
    .catch(err => {
      console.log(err);
      dispatch({ type: ERROR_TYPE, payload: err.response.data.message });
      console.log(err.response.data.message);
    });
};

export const editPlayer = player => dispatch => {
  return axios
    .put(
      "http://localhost:5000/apis/admin/editplayer/" + player.player_id,
      player,
      {
        headers: {
          Authorization: `Bearer ${localStorage.getItem("token")}`
        }
      }
    )
    .then(res => {
      dispatch({
        type: EDIT_PLAYER
      });

      alert("Edited Successfully");
    })
    .catch(err => {
      console.log(err);
    });
};

export const editTeam = team => dispatch => {
  return axios
    .put("http://localhost:5000/apis/admin/editteam/" + team.team_id, team, {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: EDIT_TEAM
      });
      // dispatch(getUsers());
      alert("Edited Successfully");
    })
    .catch(err => {
      console.log(err);
    });
};

export const deletePlayer = player_id => dispatch => {
  return axios
    .delete("http://localhost:5000/apis/admin/deleteplayer/" + player_id, {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: DELETE_PLAYER
      });
      dispatch(getPlayers());
      // alert("Player Deleted Successfully");
    })
    .catch(err => {
      console.log(err);
    });
};

export const deleteTeam = team_id => dispatch => {
  return axios
    .delete("http://localhost:5000/apis/admin/deleteteam/" + team_id, {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: DELETE_TEAM
      });
      dispatch(getAllTeams());
      // alert("Team Deleted Successfully");
    })
    .catch(err => {
      console.log(err);
    });
};

export const getPlayerSearch = searchString => dispatch => {
  return axios
    .get("http://localhost:5000/apis/admin/search/" + searchString)
    .then(res => {
      dispatch({
        type: GET_PLAYER_SEARCH,
        payload: res.data.data
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getTeamSearch = searchString => dispatch => {
  return axios
    .get("http://localhost:5000/apis/admin/searchteam/" + searchString)
    .then(res => {
      dispatch({
        type: GET_TEAM_SEARCH,
        payload: res.data.data
      });
    })
    .catch(err => {
      console.log(err);
    });
};
