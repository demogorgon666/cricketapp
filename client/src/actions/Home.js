import {
  GET_PLAYERS,
  GET_STATISTICS,
  GET_PLAYER_STATS,
  GET_TOP_BATSMAN,
  GET_NEWS,
  GET_HOME_RECENT_MATCHES
} from './Types'
import axios from 'axios'

export const getNews = () => dispatch => {
  axios
    .get('http://localhost:5000/apis/news')
    .then(res => {
      dispatch({
        type: GET_NEWS,
        payload: res.data.data
      })
      console.log("from 'then' get news")
    })

    .catch(err => console.log(err))
}

// fetching recent matches
export const getRecentMatches = () => dispatch => {
  console.log('from actions of home recent matches')
  axios
    .get('http://localhost:5000/apis/recentMatches')
    .then(res => {
      dispatch({
        type: GET_HOME_RECENT_MATCHES,
        payload: res.data.data
      })
      console.log("from 'then' get RECENT MATCHES")
    })

    .catch(err => console.log(err))
}
