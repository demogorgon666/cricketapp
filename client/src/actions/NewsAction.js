import { GET_NEWS_ID } from "./Types";
import axios from "axios";
import { id } from "date-fns/esm/locale";

export const getNewsById = id => dispatch => {
  console.log("getnews by id", id);
  return axios
    .get("http://localhost:5000/apis/News/newsbyid/" + id)
    .then(res => {
      console.log("From newsbyid actions", res.data.data);
      dispatch({
        type: GET_NEWS_ID,
        payload: res.data.data,
      });
    })
    .catch(err => {
      console.log(err);
    });
};
