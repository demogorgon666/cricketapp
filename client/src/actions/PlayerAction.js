import {
  GET_PLAYERS,
  GET_SINGLE_PLAYER,
  GET_ALL_BATSMAN,
  GET_ALL_BOWLERS,
  GET_BATSMAN_STATS,
  GET_ODI_BATSMAN_STATS,
  GET_T20_BATSMAN_STATS,
  GET_ODI_BOWLER_STATS,
  GET_T20_BOWLER_STATS,
  GET_TEST_BOWLER_STATS,
  GET_TOP_SIXES
} from "./Types";
import axios from "axios";

export const getPlayers = () => dispatch => {
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/allPlayer")
    .then(res => {
      console.log("From Player Actions", res.data.data);
      dispatch({
        type: GET_PLAYERS,
        payload: res.data.data
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getSinglePlayer = player_id => dispatch => {
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/singlePlayer/" + player_id)
    .then(res => {
      console.log(
        "From Player Actions Single player is: ",
        res.data.data[0].player_name
      );
      dispatch({
        type: GET_SINGLE_PLAYER,
        payload: res.data.data
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getBatsmen = match_type => dispatch => {
  return axios
    .post("http://localhost:5000/apis/PlayerInfo/TopBatsman", match_type)
    .then(res => {
      // console.log("From Player Actions batsman data is: ", res.data.data);
      dispatch({
        type: GET_ALL_BATSMAN,
        payload: res.data.data
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getBowlers = match_type => dispatch => {
  return axios
    .post("http://localhost:5000/apis/PlayerInfo/TopBowlers", match_type)
    .then(res => {
      // console.log("From Player Actions bowler data is: ", res.data.data);
      dispatch({
        type: GET_ALL_BOWLERS,
        payload: res.data.data
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getTopSixes = match_type => dispatch => {
  return axios
    .post("http://localhost:5000/apis/PlayerInfo/TopSixes", match_type)
    .then(res => {
      // console.log("From Player Actions bowler data is: ", res.data.data);
      dispatch({
        type: GET_TOP_SIXES,
        payload: res.data.data
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getODIBatsmanStats = player_id => dispatch => {
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/ODI-Batsman-Stats/" + player_id)
    .then(res => {
      // console.log("From Player Actions odi Batsman stats is: ", res.data.ODI);
      dispatch({
        type: GET_ODI_BATSMAN_STATS,
        payload: res.data.ODI
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getODIBowlerStats = player_id => dispatch => {
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/ODIBowlerStats/" + player_id)
    .then(res => {
      console.log(
        "From Player Actions odi BOWLER stats is: ",
        res.data.ODIBowler
      );
      dispatch({
        type: GET_ODI_BOWLER_STATS,
        payload: res.data.ODIBowler
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getTestBatsmanStats = player_id => dispatch => {
  return axios
    .get(
      "http://localhost:5000/apis/PlayerInfo/Test-Batsman-Stats/" + player_id
    )
    .then(res => {
      // console.log("From Player Actions test Batsman stats is: ", res.data.Test);
      dispatch({
        type: GET_BATSMAN_STATS,
        payload: res.data.Test
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getTestBowlerStats = player_id => dispatch => {
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/Test-Bowler-Stats/" + player_id)
    .then(res => {
      console.log(
        "From Player Actions test Batsman stats is: ",
        res.data.TestBowler
      );
      dispatch({
        type: GET_TEST_BOWLER_STATS,
        payload: res.data.TestBowler
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getT20BatsmanStats = player_id => dispatch => {
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/T20-Batsman-Stats/" + player_id)
    .then(res => {
      // console.log("From Player Actions test Batsman stats is: ", res.data.T20);
      dispatch({
        type: GET_T20_BATSMAN_STATS,
        payload: res.data.T20
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getT20BowlerStats = player_id => dispatch => {
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/T20-Bowler-Stats/" + player_id)
    .then(res => {
      console.log(
        "From Player Actions ODI BOWLER stats is: ",
        res.data.T20Bowler
      );
      dispatch({
        type: GET_T20_BOWLER_STATS,
        payload: res.data.T20Bowler
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getBatsmanStats = player_id => dispatch => {
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/Batsman-Stats/" + player_id)
    .then(res => {
      console.log("From Player Actions test Batsman stats is: ", res.data.data);
      dispatch({
        type: GET_T20_BOWLER_STATS,
        payload: res.data.data
      });
    })
    .catch(err => {
      console.log(err);
    });
};
