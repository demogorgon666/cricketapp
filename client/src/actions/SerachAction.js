import { GET_SEARCH } from "./Types";
import axios from "axios";
import Search from "../reducers/Search";

export const getSearch = (search_term, history) => dispatch => {
	console.log("actions getsearch", search_term);
	return axios
		.post("http://localhost:5000/apis/Search/search", search_term)
		.then(res => {
			console.log("From Search Actions", res.data.data);
			dispatch({
				type: GET_SEARCH,
				payload: res.data.data
			});
			console.log("actions history", history);
			console.log("res", res.data.data[0].player_id);

			history.push("/playerInfo/" + res.data.data[0].player_id);
		})
		.catch(err => {
			console.log(err);
		});
};
