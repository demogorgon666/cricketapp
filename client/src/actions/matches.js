import { GET_RECENT_MATCHES, GET_MATCHES_DATE, GET_MATCH_DETAILS_BY_ID } from './Types'
import axios from "axios";

const url = 'http://localhost:5000/api/matches';

export const getRecentMatches = (date) => dispatch => {
    return axios
        .get(url + "/recent/" + date, {
        })
        .then(res => {
            dispatch({
                type: GET_RECENT_MATCHES,
                payload: res.data.data
            });
            console.log(res.data);
        })
        .catch(err => {
            console.log(err);
        });
};

export const getMatchesDate = () => dispatch => {
    return axios
        .get(url + "/bydate", {
        })
        .then(res => {
            dispatch({
                type: GET_MATCHES_DATE,
                payload: res.data.data
            });
            console.log(res.data);
        })
        .catch(err => {
            console.log(err);
        });
};

export const getmatchdetailbyId = (id) => dispatch => {
    return axios
        .get(url + "/summary/" + id, {
        })
        .then(res => {
            dispatch({
                type: GET_MATCH_DETAILS_BY_ID,
                payload: res.data.data
            });
            console.log(res.data);
        })
        .catch(err => {
            console.log(err);
        });
};

