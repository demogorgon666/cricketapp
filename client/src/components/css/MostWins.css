.grid-class-topteam {
  width: 350px;
  border: 1px solid #707070;
  border-radius: 8px;
  margin: 15px;
  margin-left: 0px;
  /* position: fixed; */
}

.p-topteam {
  width: 98%;
  text-align: center;
  font-size: 28px;
  color: #2c3e50;
}

.cards {
  display: flex;
  margin-left: 15%;
  margin-right: 15%;
  margin-bottom: 5%;
}
.cardtest {
  color: #707070;
  background-color: #16a085;
  color: white;
  width: 100%;
  cursor: pointer;
}

.cardtest p {
  color: #ffffff;
}

.cardodi {
  color: #707070;
  border: 1px solid #16a085;
  width: 100%;
  cursor: pointer;
}

.cardt20 {
  color: #707070;
  border: 1px solid #16a085;
  width: 100%;
  cursor: pointer;
}

.cardtest:hover {
  background-color: #16a085;
  color: white;
}
.cardodi:hover {
  background-color: #16a085;
  color: white;
}
.cardt20:hover {
  background-color: #16a085;
  color: white;
}

.cardtest:hover .p-card {
  color: white;
}
.cardodi:hover .p-card {
  color: white;
}
.cardt20:hover .p-card {
  color: white;
}

.p-card {
  text-align: center;
}

.list-wins {
  text-align: center;
  display: flex;
  margin: 10px 10px 10px 10px;
}

.list-wins:not(:last-child) {
  border-bottom: 1px solid #707070;
}

.img-card {
  width: 60px;
  height: 60px;
  margin-right: 5%;
  border-radius: 50%;
}
.p-team-name {
  font-size: 20px;
  margin-right: auto;
  text-align: left;
}

.h2-team-position {
  font-size: 20px;
  margin-left: auto;
  margin-right: 18px;
}

.hr-team-card {
  margin-left: 5%;
  margin-right: 5%;
}
