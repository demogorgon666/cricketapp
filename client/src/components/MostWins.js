import React, { Component } from "react";
import { connect } from "react-redux";
import "./css/MostWins.css";
import logo from "./images/indialogo.jpg";
import { getRanks } from "../actions/Teams";

export class MostWins extends Component {
  state = {
    testClick: false,
    odiClick: false,
    t20Click: true,
    iplClick: false
  };

  componentDidMount() {
    console.log("ranking gender - ", this.props.gender);
    console.log(this.props);
    let ranking = {
      match_type: "T20",
      competition: "others"
    };
    this.props.getRanks(this.props.gender, ranking);
  }

  onClickTest = () => {
    this.setState({
      testClick: true,
      odiClick: false,
      t20Click: false,
      iplClick: false
    });
    let ranking = {
      match_type: "Test",
      competition: "others"
    };
    this.props.getRanks(this.props.gender, ranking);
  };

  onClickOdi = () => {
    this.setState({
      testClick: false,
      odiClick: true,
      t20Click: false,
      iplClick: false
    });
    let ranking = {
      match_type: "ODI",
      competition: "others"
    };
    this.props.getRanks(this.props.gender, ranking);
  };

  onClickT20 = () => {
    this.setState({
      testClick: false,
      odiClick: false,
      t20Click: true,
      iplClick: false
    });
    let ranking = {
      match_type: "T20",
      competition: "others"
    };
    this.props.getRanks(this.props.gender, ranking);
  };

  onClickIpl = () => {
    this.setState({
      testClick: false,
      odiClick: false,
      t20Click: false,
      iplClick: true
    });
    let ranking = {
      match_type: "T20",
      competition: "IPL"
    };
    this.props.getRanks(this.props.gender, ranking);
  };

  render() {
    return (
      <div>
        <div className="grid-class-topteam">
          <div className="info-topteam">
            <p className="p-topteam">Most Wins</p>
            <div className="centered">
              <div className="cards">
                <div
                  style={{ borderRadius: "8px 0px 0px 8px" }}
                  className={this.state.t20Click ? "cardtest" : "cardodi"}
                  onClick={this.onClickT20}
                >
                  <p className="p-card">IT20</p>
                </div>
                <div
                  className={this.state.odiClick ? "cardtest" : "cardodi"}
                  onClick={this.onClickOdi}
                >
                  <p className="p-card">ODI</p>
                </div>
                <div
                  className={this.state.testClick ? "cardtest" : "cardt20"}
                  onClick={this.onClickTest}
                >
                  <p className="p-card">Test</p>
                </div>
                <div
                  style={{ borderRadius: "0px 8px 8px 0px" }}
                  className={this.state.iplClick ? "cardtest" : "cardt20"}
                  onClick={this.onClickIpl}
                >
                  <p className="p-card">IPL</p>
                </div>
              </div>
            </div>
            <div>
              {this.props.ranks.map(ranks => (
                <div className="list-wins">
                  <img src={logo} className="img-card" />
                  <p className="p-team-name">{ranks.team_name}</p>
                  <h2 className="h2-team-position">{ranks.count}</h2>
                </div>
              ))}
            </div>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateTostate = state => ({
  ranks: state.TeamsReducer.ranks
});

export default connect(
  mapStateTostate,
  { getRanks }
)(MostWins);
