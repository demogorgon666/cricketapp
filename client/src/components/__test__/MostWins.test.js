import React from "react";
import { shallow } from "enzyme";
import { MostWins } from "../MostWins";
import { getRanks } from "../../actions/teams";

var ranks = [];
const wrapper = shallow(<MostWins getRanks={getRanks} ranks={ranks} />);

describe("Test MostWins Component", () => {
  it("render the component", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should have links for Test,ODI,T20 and IPL", () => {
    expect(
      wrapper
        .find("p")
        .at(1)
        .text()
    ).toBe("Test");

    expect(
      wrapper
        .find("p")
        .at(2)
        .text()
    ).toBe("ODI");

    expect(
      wrapper
        .find("p")
        .at(3)
        .text()
    ).toBe("IT20");

    expect(
      wrapper
        .find("p")
        .at(4)
        .text()
    ).toBe("IPL");
  });
  it("should check if the tabs are getting clicked ", () => {
    wrapper
      .find("div")
      .at(0)
      .simulate("click");
  });
  it("should change the state of testClicked to true when test tab is clicked", () => {
    const test = wrapper.find("div").at(5);
    test.simulate("click");
    expect(wrapper.state().testClick).toBe(true);
    expect(wrapper.state().odiClick).toBe(false);
    expect(wrapper.state().t20Click).toBe(false);
    expect(wrapper.state().iplClick).toBe(false);
  });

  it("should change the state of odiclicked to true when odi tab is clicked", () => {
    const odi = wrapper.find("div").at(6);
    odi.simulate("click");
    expect(wrapper.state().testClick).toBe(false);
    expect(wrapper.state().odiClick).toBe(true);
    expect(wrapper.state().t20Click).toBe(false);
    expect(wrapper.state().iplClick).toBe(false);
  });

  it("should change the state of t20Click to true when IT20 tab is clicked", () => {
    const t20 = wrapper.find("div").at(7);
    t20.simulate("click");
    expect(wrapper.state().odiClick).toBe(false);
    expect(wrapper.state().testClick).toBe(false);
    expect(wrapper.state().t20Click).toBe(true);
    expect(wrapper.state().iplClick).toBe(false);
  });
  it("should change the state of iplClick to true when IPL tab is clicked", () => {
    const ipl = wrapper.find("div").at(8);
    ipl.simulate("click");
    expect(wrapper.state().odiClick).toBe(false);
    expect(wrapper.state().testClick).toBe(false);
    expect(wrapper.state().t20Click).toBe(false);
    expect(wrapper.state().iplClick).toBe(true);
  });
  it("should have no br tags", () => {
    expect(wrapper.find("br").length).toBe(0);
  });
  it("renders", () => {
    expect(wrapper.exists()).toBe(true);
  });
  it("should have className = 'grid-class-topteam' exacty 1", () => {
    expect(wrapper.find(".grid-class-topteam").length).toBe(1);
  });
  it("should have className = 'cards' exacty 1", () => {
    expect(wrapper.find(".cards").length).toBe(1);
  });
  it("should have className = 'cardtest' exacty 1", () => {
    expect(wrapper.find(".cardtest").length).toBe(1);
  });
  it("should have className = 'cardodi' exacty 2", () => {
    expect(wrapper.find(".cardodi").length).toBe(2);
  });
  it("should have className = 'cardt20' exacty 1", () => {
    expect(wrapper.find(".cardt20").length).toBe(1);
  });
  it("should have className = 'p-card' exacty 4", () => {
    expect(wrapper.find(".p-card").length).toBe(4);
  });
});
