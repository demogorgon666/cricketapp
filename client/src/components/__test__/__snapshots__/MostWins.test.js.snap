// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MostWins Component render the component 1`] = `
<div>
  <div
    className="grid-class-topteam"
  >
    <div
      className="info-topteam"
    >
      <p
        className="p-topteam"
      >
        Most Wins
      </p>
      <div
        className="centered"
      >
        <div
          className="cards"
        >
          <div
            className="cardtest"
            onClick={[Function]}
            style={
              Object {
                "borderRadius": "8px 0px 0px 8px",
              }
            }
          >
            <p
              className="p-card"
            >
              Test
            </p>
          </div>
          <div
            className="cardodi"
            onClick={[Function]}
          >
            <p
              className="p-card"
            >
              ODI
            </p>
          </div>
          <div
            className="cardt20"
            onClick={[Function]}
          >
            <p
              className="p-card"
            >
              IT20
            </p>
          </div>
          <div
            className="cardt20"
            onClick={[Function]}
            style={
              Object {
                "borderRadius": "0px 8px 8px 0px",
              }
            }
          >
            <p
              className="p-card"
            >
              IPL
            </p>
          </div>
        </div>
      </div>
      <div />
    </div>
  </div>
</div>
`;
